chrome/install.js

```diff
@@ -74,6 +74,12 @@ function onLoad() {
 				if(success) wizard.advance();
 			}
 		});
+	} else {
+		checkMacJDK().then(function() {
+			wizard.getPageById("intro").next = "openoffice-installations";
+		}).catch(function() {
+			wizard.getPageById("intro").next = "jdk-required";
+		})
```

Comment:
Certainly is more reasonable. Although annoyingly, `Zotero.Utilities.Internal.exec()` actually [does this awkwardly](https://github.com/zotero/zotero/blob/22eab3e09d57c378a91d20a5973b4654975d26ea/chrome/content/zotero/xpcom/utilities_internal.js#L491-L496) too.